mibidata/tiff.py

```diff
@@ -82,12 +84,17 @@ def write(filename, image, sed=None, optical=None, ranges=None,
         raise ValueError('Image coordinates and size must not be None.')
     if image.masses is None or image.targets is None:
         raise ValueError('Image channels must contain both masses and targets.')
-    if np.issubdtype(image.data.dtype, np.integer) and not write_float:
-        range_dtype = 'I'
+    if write_dtype and not write_dtype in ['float', 'int', np.float, np.uint16]:
+        raise ValueError('Invalid dtype specification.')
+    if not write_dtype:
+        range_dtype = 'I' if\
+            np.issubdtype(image.data.dtype, np.integer) else 'd'
     else:
-        range_dtype = 'd'
+        range_dtype = 'I' if write_dtype in ['int', np.uint16] else 'd'
+
     if ranges is None:
-        ranges = [(0, m) for m in image.data.max(axis=(0, 1))]
+        dtype_conversion = int if range_dtype == 'I' else float
+        ranges = [(0, dtype_conversion(m)) for m in image.data.max(axis=(0, 1))]
```

Comment:
Yes, otherwise the add tag function in the underlying TIFF writer complains the value is not of the correct type. Not sure how it worked before this change...

Reply:
I still don't think you need to do the conversion here if you use the data you've already converted:

```
if ranges is None:
  ranges = [(0, m) for m in to_save.max(axis=(0, 1))]